src/realm/object-store/impl/list_notifier.cpp

```diff
@@ -49,8 +49,9 @@ void ListNotifier::attach(CollectionBase const& src)
     auto& tr = transaction();
     try {
         auto obj = tr.get_table(src.get_table()->get_key())->get_object(src.get_owner_key());
-        // FIXME: Use path for list when supporting notifications for nested collections
-        m_list = obj.get_collection_ptr(src.get_col_key());
+        auto path = src.get_short_path();
+        auto full = src.get_path().path_from_top;
```

Comment:
Unless I'm missing something about Path, this doesn't work when the source and destination transactions are at different versions, which is something that this needs to support.

Reply:
Yes, this should be fixed as well. Should be based on StablePath.

Reply:
@tgoyne it is not clear to me how we could write a test for this. I have started on a test, but it does not fail. Can you help?